### DISCUSSION TITLE
Version for FreeBSD

### DISCUSSION THREAD

**Discussion by hinqiwame** [ppy/osu#22278] [Ideas] [answered]
Is it possible?

**Comment by bdach** [comment, marked as answer]
Technically possible? Maybe. As long as our upstreams all support *BSD, and as long as there are graphics drivers for *BSD that can actually do opengl (or other graphics APIs, once they come).

Will there be an official release for *BSD with all of the online functionality? Probably not unless the current linux appimage can be made to work out of the box.

Are we going to actively work on it at some point? No.

The interest for *BSD gaming is minuscule. I think you need to be real with yourself as to how gaming-viable that OS is.

**Comment by 64ArthurAraujo** [comment]
From what i've tested, with some tinkering it is possible to run the template games of the osu!framework almost flawlessly
 
![shot-2023-02-28_19-50-19](https://user-images.githubusercontent.com/90941580/221999876-75a95958-d03f-4435-8d97-43d9dcd074de.jpg)

Now for lazer the issue seems to be the realm database that have some problems with its .so files. To get this working is a very boring process, i would only do it if i really wanted to use BSD as a daily driver and osu! was the only game that i played. I will keep trying to make it work tho just for the sake of it

**Comment by Axilotl17** [comment]
for now you can build manually https://github.com/ppy/osu#developing-osu

**Comment by s0m3sushi** [comment]
So far I was able to compile and start the game by compiling it with ``dotnet`` binaries from [Thefrank](https://github.com/Thefrank/dotnet-freebsd-crossbuild/releases), and run it with OpenGL detected thanks to ``nv-sglrun`` script provided by ``libc6-shim``. All that's left is Realms wrapper library. There hasn't been any successful ports of that for FreeBSD yet.
![2023-12-23_1341x321](https://github.com/ppy/osu/assets/30410170/988a9608-fb68-4d39-8627-e33012c5b96d)

**Reply by oblivcheck** [reply]
how did you build it？
`System.PlatformNotSupportedException: Operating system could not be detected correctly.`

I use [v6.0.416](https://github.com/Thefrank/dotnet-freebsd-crossbuild/releases/tag/v6.0.416)

**Reply by s0m3sushi** [reply]
I got it to build by applying this simple patch to ``osu-framework`` https://github.com/ppy/osu-framework/commit/ac2bcd391fcda9222bcbeb1f5a23c5e930360b7e

**Comment by s0m3sushi** [comment]
Update: I managed to mitigate Realms wrapper library problem by manually placing ``librealm-wrappers.so`` into ``/compat/linux/lib64/``. Now there's a new problem, and it's ``libbass``.
![2024-01-07_1344x347](https://github.com/ppy/osu/assets/30410170/0cab2fa8-afeb-4fd3-bd94-ec9f23c511f2)

**Reply by akai-hana** [reply]
any updates on this?

**Reply by s0m3sushi** [reply]
``libbass`` does not support FreeBSD, and I haven't figured out how to get it to work with Linuxulator, so we are stuck right now.

**Reply by darkralts** [reply]
> `libbass` does not support FreeBSD, and I haven't figured out how to get it to work with Linuxulator, so we are stuck right now.

perhaps theres a way to port libbass to freebsd? altough i never even heard of libbass before.

**Reply by bdach** [reply]
It's closed source, so good luck?

**Comment by darkralts** [comment]
why is osu open source edition using proprietary things? is there an
alternative?

**Reply by bdach** [reply]
> why is osu open source edition using proprietary things?

because BASS works pretty well cross-platform all things considered

> is there an alternative?

https://github.com/ppy/osu-framework/issues/2784
tl;dr: no (not yet?)
